Error when 'grantReadWriteData' on a DDB table to an IGrantable Construct

Repo with reproduction: https://github.com/ryancormack/cdktf-ddb-bug

### Description

Using a DDB table's `grantReadWriteData` method to create the permission set to allow a Lambda function to read and write the data results in an error.
`grantReadData`
`grantWriteData`
`grantFullAccess`
All exhibit the same behaviour

#### Expected result

A Terraform json document with a lambda policy that permits access to the DDB table.

#### Actual

```
⠧  Synthesizing
[2022-09-09T17:18:28.723] [ERROR] default - /Users/ryan.cormack/code/ryancormack/cdktf-ddb-bug/node_modules/@cdktf/aws-cdk/src/aws-adapter.ts:358
        throw new Error(`unable to resolve pseudo reference ${ref}`);
              ^

[2022-09-09T17:18:28.725] [ERROR] default - Error: unable to resolve pseudo reference AWS::NoValue
    at TerraformHost.resolvePseudo (/Users/ryan.cormack/code/ryancormack/cdktf-ddb-bug/node_modules/@cdktf/aws-cdk/src/aws-adapter.ts:358:15)
    at TerraformHost.resolveRef (/Users/ryan.cormack/code/ryancormack/cdktf-ddb-bug/node_modules/@cdktf/aws-cdk/src/aws-adapter.ts:364:19)
    at TerraformHost.processIntrinsics (/Users/ryan.cormack/code/ryancormack/cdktf-ddb-bug/node_modules/@cdktf/aws-cdk/src/aws-adapter.ts:293:19)
    at /Users/ryan.cormack/code/ryancormack/cdktf-ddb-bug/node_modules/@cdktf/aws-cdk/src/aws-adapter.ts:288:34
    at Array.map (<anonymous>)
    at TerraformHost.processIntrinsics (/Users/ryan.cormack/code/ryancormack/cdktf-ddb-bug/node_modules/@cdktf/aws-cdk/src/aws-adapter.ts:288:18)
    at TerraformHost.processIntrinsics (/Users/ryan.cormack/code/ryancormack/cdktf-ddb-bug/node_modules/@cdktf/aws-cdk/src/aws-adapter.ts:307:24)
    at /Users/ryan.cormack/code/ryancormack/cdktf-ddb-bug/node_modules/@cdktf/aws-cdk/src/aws-adapter.ts:288:34
    at Array.map (<anonymous>)
ERROR: cdktf encountered an error while synthesizing

Synth command: npx ts-node main.ts
Error:         non-zero exit code 1
```

#### Environment

```
% cdktf --version
0.12.2
% terraform --version
Terraform v1.2.1
on darwin_arm64

Your version of Terraform is out of date! The latest version
is 1.2.9. You can update by downloading from https://www.terraform.io/downloads.html
% node --version
v16.15.0
```

--- a/src/aws-adapter.ts
+++ b/src/aws-adapter.ts
@@ -354,6 +354,10 @@ class TerraformHost extends Construct {
         return this.awsCallerIdentity.accountId;
       }
 
+      case "AWS::NoValue": {
+        return undefined;
+      }
+
       default:
         throw new Error(`unable to resolve pseudo reference ${ref}`);
     }

